server/api.js

```
@@ -1 +1,57 @@
-// functon for getting all data
\ No newline at end of file
+import { SERVER_URL, POST, DELETE, PUT } from "./constants.js";
+
+export const api = async (method, payload, endpoint = `${SERVER_URL}`) => {
+  let config = {};
+
+  if (method) {
+    config = {
+      method,
+      headers: {
+        'Content-Type' : 'application/json',
+      }
+    }
+
+    if (method === POST || method === PUT) {
+      config.body = JSON.stringify(payload.body);
+    }
+
+    if (method === DELETE || method === PUT) {
+      endpoint = `${endpoint}/${payload.id}`;
+    }    
+  }
+
+  try {
+
+    const response = await fetch(endpoint, config);
+
+    if (response.ok) {
```

Отлично, но обработку ошибок можно улучшить. Проверяй не на успех, а на неудачу с выбросом объекта ошибки: если запрос неуспешен, функция сразу выбрасывает ошибку, не выполняя лишний код. Это соответствует принципу "fail fast". Также уменьшается вероятность забыть обработку.

---

Исправлено